### PR TITLE
Double Chance markets

### PR DESCRIPTION
**Pull request by hinkypunks**

### CODE REVIEW

File: contracts/SportMarkets/SportsAMM.sol [hunk 1]
```diff
@@ -96,20 +96,20 @@ contract SportsAMM is Initializable, ProxyOwned, PausableUpgradeable, ProxyReent
...
-    /// @return The address of Parlay AMM
-    address public parlayAMM;
-
     /// @return The address of Apex Consumer
     address public apexConsumer;
 
-    /// @return maximum supported discount in percentage on sUSD purchases with different collaterals
-    uint public maxAllowedPegSlippagePercentage;
+    /// @return The address of Parlay AMM
+    address public parlayAMM;
```
Comment (grujawork) [inline contracts/SportMarkets/SportsAMM.sol:112]: don't change order of properties due to storage :)

i know it is different on `opt-goerli` and `opt-main` so only change order locally without pushing

**Review by grujawork — CHANGES_REQUESTED** [review]

File: contracts/SportMarkets/SportPositions/SportPositionalMarket.sol [hunk 2]
```diff
@@ -339,8 +367,10 @@ contract SportPositionalMarket is OwnedWithInit, ISportPositionalMarket {
 
         _mint(msg.sender, value);
 
-        _incrementDeposited(value);
-        _manager().transferSusdTo(msg.sender, address(this), value);
+        if (!isDoubleChance) {
```
Comment (danijelthales) [inline contracts/SportMarkets/SportPositions/SportPositionalMarket.sol:370]: It is incorrect to mint all positions for double chance. Rather, AMM should only mint the position it is selling.

**Review by danijelthales — CHANGES_REQUESTED** [review]

File: contracts/SportMarkets/Rundown/TherundownConsumer.sol [hunk 3]
```diff
@@ -473,7 +473,11 @@ contract TherundownConsumer is Initializable, ProxyOwned, ProxyPausable {
 
             uint index = sportsManager.numActiveMarkets() - 1;
             if (!isSportTwoPositionsSport(sportsIdPerGame[_gameId])) {
```
Comment (danijelthales) [inline contracts/SportMarkets/Rundown/TherundownConsumer.sol:475]: we might have a sport that is more than two positional that is not soccer (no double chance).
Might be fine for now, but an alternative is to have a list of sportIDs that support double chance

File: contracts/SportMarkets/Rundown/TherundownConsumer.sol [hunk 4]
```diff
@@ -473,7 +473,11 @@ contract TherundownConsumer is Initializable, ProxyOwned, ProxyPausable {
...
-                index = sportsManager.numActiveMarkets() - 2; // double chance market is the last one of active markets
+                if (sportsManager.isDoubleChanceSupported()) {
+                    index = sportsManager.numActiveMarkets() - 4; // three double chance markets are following their parent
+                } else {
+                    index = sportsManager.numActiveMarkets() - 1;
+                }
             }
             address marketAddress = sportsManager.getActiveMarketAddress(index);
```
Comment (danijelthales) [inline contracts/SportMarkets/Rundown/TherundownConsumer.sol:480]: I realize this is not part of this PR, but I dont get why the market address is not just returned via `createMarket` method, rather than use this index of the array logic.

File: contracts/SportMarkets/SportPositions/SportPositionalMarket.sol [hunk 5]
```diff
@@ -154,7 +154,13 @@ contract SportPositionalMarket is OwnedWithInit, ISportPositionalMarket {
     }
 
     function _expired() internal view returns (bool) {
-        return isResolved && (times.expiry < block.timestamp || deposited == 0);
+        return resolved && (times.expiry < block.timestamp || deposited == 0);
+    }
+
+    function _isPaused() internal view returns (bool) {
+        if (isDoubleChance) return parentMarket.paused();
```
Comment (danijelthales) [inline contracts/SportMarkets/SportPositions/SportPositionalMarket.sol:161]: rewrite as ternary expression
return `isDoubleChange?parentMarket.paused():paused`

File: contracts/SportMarkets/SportPositions/SportPositionalMarket.sol [hunk 6]
```diff
@@ -374,22 +382,23 @@ contract SportPositionalMarket is OwnedWithInit, ISportPositionalMarket {
...
         }
     }
 
-    function burnOptionsMaximum() external override {
-        require(msg.sender == sportsAMM, "Invalid burner");
-        _burnOptions(msg.sender, _getMaximumBurnable(msg.sender));
-    }
-
     function burnOptions(uint amount) external override {
+        require(!isDoubleChance, "Not supported for double chance markets");
```
Comment (danijelthales) [inline contracts/SportMarkets/SportPositions/SportPositionalMarket.sol:401]: not just for double chance, deprecate this in general now that sell is disabled

File: contracts/SportMarkets/SportPositions/SportPositionalMarketManager.sol [hunk 7]
```diff
@@ -146,6 +146,16 @@ contract SportPositionalMarketManager is Initializable, ProxyOwned, ProxyPausabl
         }
     }
 
+    function getDoubleChanceMarkets(address market) external view returns (address[] memory) {
```
Comment (danijelthales) [inline contracts/SportMarkets/SportPositions/SportPositionalMarketManager.sol:149]: why is this method needed since the variable is already public? 
`  mapping(address => address[]) public doubleChanceMarkets;`

**Review by danijelthales — CHANGES_REQUESTED** [review]
